cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -3896,6 +3896,41 @@ INSTANTIATE_TEST_SUITE_P(WriteDictionary, TestArrowWriteDictionary,
                          ::testing::Values(ParquetDataPageVersion::V1,
                                            ParquetDataPageVersion::V2));
 
+TEST(TestArrowWriteNestedDictionary, TestNullCount) {
+  std::shared_ptr<::arrow::Array> dictionary = ArrayFromJSON(::arrow::utf8(), R"(["z"])");
```

Comment:
any reason to use dictionary here?

Reply:
If you are asking why I added a dictionary specific test, it is to test the changes to WriteArrowDictionary; the other tests were not sufficient.